android-components/components/lib/state/src/main/java/mozilla/components/lib/state/internal/StoreThreadFactory.kt

```diff
@@ -51,7 +51,8 @@ internal class StoreThreadFactory(
         }
 
         throw IllegalThreadStateException(
-            "Expected `store` thread, but running on thread `${currentThread.name}`. Leaked MiddlewareContext?",
+            "Expected `store` thread, but running on thread `${currentThread.name}`. " +
+                "Leaked MiddlewareContext or did you mean to use `MiddlewareContext.dispatch`?",
```

Comment:
Thank you for adding more info to this error! 🤩

We may want to also consider adding `or Store.dispatch?` For my case it was [`Store.dispatch`](https://searchfox.org/mozilla-mobile/rev/25e5c254fd39d3d1a3ef941a6aa29ca4f34b8400/firefox-android/android-components/components/lib/state/src/main/java/mozilla/components/lib/state/Store.kt#94) that I needed to call. This is the [comment](https://searchfox.org/mozilla-mobile/rev/25e5c254fd39d3d1a3ef941a6aa29ca4f34b8400/firefox-android/android-components/components/lib/state/src/main/java/mozilla/components/lib/state/Middleware.kt#44-45) I wish I saw earlier. 

```suggestion
                "Leaked MiddlewareContext or did you mean to use `MiddlewareContext.dispatch or Store.dispatch`?",
```

Reply:
Oops, `context.store.dispatch` is what I meant, thanks for the catch!